src/lua/compat.lua

```diff
@@ -287,6 +294,13 @@ local options = {
         brief = WAL_CLEANUP_DELAY_DEPRECATION_BRIEF,
         action = function() end,
     },
+    box_cfg_replication_synchro_timeout = {
```

Comment:
Nit: I don't think that the `box_cfg_` prefix is mandatory for such compat options, especially in the light of the idea that the cluster configuration is/will be the primary one, where the option is just `replication.synchro_timeout`. Just `replication_synchro_timeout` or `replication_synchro_timeout_meaning` looks better for me for the compat option name.

Reply:
```diff
diff --git a/changelogs/unreleased/gh-7486-prohibit-roll-back-due-to-replication_synchro_timeout.md b/changelogs/unreleased/gh-7486-prohibit-roll-back-due-to-replication_synchro_timeout.md
index 205f5bece..0ef0fa34f 100644
--- a/changelogs/unreleased/gh-7486-prohibit-roll-back-due-to-replication_synchro_timeout.md
+++ b/changelogs/unreleased/gh-7486-prohibit-roll-back-due-to-replication_synchro_timeout.md
@@ -1,16 +1,14 @@
 ## feature/replication
 
-This patch is designed to prohibit automatic transaction rollback due to
-`replication_synchro_timeout`. To do this, the following was done:
-* A new compat option `box_cfg_replication_synchro_timeout` has been added.
-  This option determines whether the `replication_synchro_timeout` option rolls
+* A new compat option `compat.replication_synchro_timeout` has been added.
+  This option determines whether the `replication.synchro_timeout` option rolls
   back transactions. When set to 'new', transactions are not rolled back due to
-  a timeout. In this mode `replication_synchro_timeout` is used to wait
+  a timeout. In this mode `replication.synchro_timeout` is used to wait
   confirmation in promote/demote and gc-checkpointing. If 'old' is set, the
   behavior is no different from what it was before this patch appeared.
-* A new `replication_synchro_queue_max_size` option limits the number of
+* A new `replication.synchro_queue_max_size` option limits the number of
   transactions in the master synchronous queue.
-  `replication_synchro_queue_max_size` is measured in the number of bytes to be
+  `replication.synchro_queue_max_size` is measured in the number of bytes to be
   written (0 means unlimited, which was the default behavior before).
   Currently, this option defaults to 16 megabytes.
   (gh-7486)
diff --git a/src/box/lua/config/instance_config.lua b/src/box/lua/config/instance_config.lua
index 204943411..d5b870d38 100644
--- a/src/box/lua/config/instance_config.lua
+++ b/src/box/lua/config/instance_config.lua
@@ -2360,12 +2360,6 @@ return schema.new('instance_config', schema.record({
         }, {
             default = 'new',
         }),
-        box_cfg_replication_synchro_timeout = schema.enum({
-            'old',
-            'new',
-        }, {
-            default = 'old',
-        }),
         sql_seq_scan_default = schema.enum({
             'old',
             'new',
@@ -2462,6 +2456,12 @@ return schema.new('instance_config', schema.record({
         }, {
             default = 'old',
         }),
+        replication_synchro_timeout = schema.enum({
+            'old',
+            'new',
+        }, {
+            default = 'old',
+        }),
     }),
     -- Instance labels.
     labels = schema.map({
diff --git a/src/lua/compat.lua b/src/lua/compat.lua
index ae599312c..fb28862bf 100644
--- a/src/lua/compat.lua
+++ b/src/lua/compat.lua
@@ -156,12 +156,12 @@ be retained by persistent WAL GC.
 https://tarantool.io/compat/wal_cleanup_delay_deprecation
 ]]
 
-local BOX_CFG_REPLICATION_SYNCHRO_TIMEOUT_COMPAT_BRIEF = [[
+local REPLICATION_SYNCHRO_TIMEOUT_COMPAT_BRIEF = [[
 Determines whether the replication_synchro_timeout option rolls back
 transactions or it only used to wait confirmation in promote/demote and
 gc-checkpointing.
 
-https://tarantool.io/compat/box_cfg_replication_synchro_timeout
+https://tarantool.io/compat/replication_synchro_timeout
 ]]
 
 -- Returns an action callback that toggles a tweak.
@@ -294,10 +294,10 @@ local options = {
         brief = WAL_CLEANUP_DELAY_DEPRECATION_BRIEF,
         action = function() end,
     },
-    box_cfg_replication_synchro_timeout = {
+    replication_synchro_timeout = {
         default = 'old',
         obsolete = nil,
-        brief = BOX_CFG_REPLICATION_SYNCHRO_TIMEOUT_COMPAT_BRIEF,
+        brief = REPLICATION_SYNCHRO_TIMEOUT_COMPAT_BRIEF,
         action = tweak_action(
             'replication_synchro_timeout_rollback_enabled', true, false),
     },
diff --git a/test/config-luatest/cluster_config_schema_test.lua b/test/config-luatest/cluster_config_schema_test.lua
index 450407c4c..fc04a72e6 100644
--- a/test/config-luatest/cluster_config_schema_test.lua
+++ b/test/config-luatest/cluster_config_schema_test.lua
@@ -337,7 +337,7 @@ g.test_defaults = function()
             box_cfg_replication_sync_timeout = 'new',
             box_consider_system_spaces_synchronous = 'old',
             box_error_serialize_verbose = 'old',
-            box_cfg_replication_synchro_timeout = 'old',
+            replication_synchro_timeout = 'old',
             sql_seq_scan_default = 'new',
             fiber_slice_default = 'new',
             box_info_cluster_meaning = 'new',
diff --git a/test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout.lua b/test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout_test.lua
similarity index 93%
rename from test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout.lua
rename to test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout_test.lua
index 54767b0e7..63c2d99ef 100644
--- a/test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout.lua
+++ b/test/replication-luatest/gh_7486_prohibit_roll_back_due_to_replication_synchro_timeout_test.lua
@@ -28,7 +28,7 @@ g.test_replication_synchro_timeout_rolls_back_if_old = function(cg)
             replication_synchro_quorum = 2,
             replication_synchro_timeout = 0.01,
         }
-        require('compat').box_cfg_replication_synchro_timeout = 'old'
+        require('compat').replication_synchro_timeout = 'old'
         t.assert_error_msg_content_equals(
             'Quorum collection for a synchronous transaction is timed out',
             box.space.test.insert, box.space.test, {1})
@@ -46,7 +46,7 @@ g.test_replication_synchro_timeout_does_not_roll_back_if_new = function(cg)
             replication_synchro_timeout = 0.01,
         }
         local fiber = require('fiber')
-        require('compat').box_cfg_replication_synchro_timeout = 'new'
+        require('compat').replication_synchro_timeout = 'new'
         local f = fiber.create(function() box.space.test:insert{1} end)
         f:set_joinable(true)
         t.helpers.retrying({timeout = wait_timeout}, function()
diff --git a/test/replication-luatest/gh_7486_replication_synchro_queue_max_size_test.lua b/test/replication-luatest/gh_7486_replication_synchro_queue_max_size_test.lua
index b52f54bc2..3582d6439 100644
--- a/test/replication-luatest/gh_7486_replication_synchro_queue_max_size_test.lua
+++ b/test/replication-luatest/gh_7486_replication_synchro_queue_max_size_test.lua
@@ -17,41 +17,28 @@ local function server_wait_synchro_queue_len_is_equal(server, expected)
 end
 
 g.before_all(function(cg)
-    cg.cluster = cluster:new({})
     cg.master_cfg = {
-        replication = {
-          server.build_listen_uri('master', cg.cluster.id),
-          server.build_listen_uri('replica', cg.cluster.id),
-        },
-        election_mode = 'candidate',
-        replication_synchro_quorum = 2,
+        election_mode = 'manual',
+        replication_synchro_quorum = 1,
         replication_synchro_timeout = 100000,
         replication_timeout = 0.1,
         election_fencing_mode = 'off',
     }
-    cg.master = cg.cluster:build_and_add_server({
+    cg.master = server:new({
         alias = 'master',
         box_cfg = cg.master_cfg,
     })
-    cg.replica_cfg = table.copy(cg.master_cfg)
-    cg.replica_cfg.election_mode = 'voter'
-    cg.replica = cg.cluster:build_and_add_server({
-        alias = 'replica',
-        box_cfg = cg.replica_cfg
-    })
-    cg.cluster:start()
-    g.cluster:wait_for_fullmesh()
-    cg.master:wait_until_election_leader_found()
-    cg.replica:wait_until_election_leader_found()
-
+    cg.master:start()
     cg.master:exec(function()
+        box.ctl.promote()
+        box.ctl.wait_rw()
         box.schema.space.create('test', {is_sync = true})
         box.space.test:create_index('pk')
     end)
 end)
 
 g.after_all(function(cg)
-    cg.cluster:drop()
+    cg.master:drop()
 end)
 
 g.before_each(function(cg)
@@ -68,7 +55,7 @@ end)
 
 g.test_master_synchro_queue_limited = function(cg)
     cg.master:exec(function()
-        box.cfg{ replication_synchro_quorum = 3, }
+        box.cfg{ replication_synchro_quorum = 2, }
         rawset(_G, "f", require('fiber')
             .new(box.space.test.insert, box.space.test, {1}))
         _G.f:set_joinable(true)
@@ -78,7 +65,7 @@ g.test_master_synchro_queue_limited = function(cg)
         t.assert_error_msg_content_equals(
             'The synchronous transaction queue is full',
             box.space.test.insert, box.space.test, {2})
-        box.cfg{ replication_synchro_quorum = 2, }
+        box.cfg{ replication_synchro_quorum = 1, }
         local ok, _ = _G.f:join()
         t.assert(ok)
     end, {wait_timeout})
@@ -88,7 +75,7 @@ g.test_max_size_update_dynamically = function(cg)
     cg.master:exec(function()
         box.cfg{
             replication_synchro_queue_max_size = 0,
-            replication_synchro_quorum = 3,
+            replication_synchro_quorum = 2,
         }
         rawset(_G, 'fibers_storage', {})
         rawset(_G, 'num_fibers', 3)
@@ -105,7 +92,7 @@ g.test_max_size_update_dynamically = function(cg)
         t.assert_error_msg_content_equals(
             'The synchronous transaction queue is full',
             box.space.test.insert, box.space.test, {0})
-        box.cfg{ replication_synchro_quorum = 2, }
+        box.cfg{ replication_synchro_quorum = 1, }
         for i = 1, _G.num_fibers do
             local ok, _ = _G.fibers_storage[i]:join()
             t.assert(ok)
@@ -121,7 +108,7 @@ g.test_recovery_with_small_max_size = function(cg)
     cg.master:exec(function(wait_timeout)
         box.cfg{
             replication_synchro_queue_max_size = 0,
-            replication_synchro_quorum = 3,
+            replication_synchro_quorum = 2,
         }
         local lsn = box.info.lsn + 1000
         for key = 1, 1000 do
@@ -145,6 +132,7 @@ g.test_recovery_with_small_max_size = function(cg)
     server_wait_synchro_queue_len_is_equal(cg.master, 1000)
     cg.master:exec(function()
         box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        box.ctl.promote()
         box.ctl.wait_rw()
         t.assert_equals(box.space.test:len(), 1000)
     end)
@@ -160,14 +148,14 @@ end
 g.test_size_is_updated_correctly_after_rollback = function(cg)
     cg.master:exec(function()
         box.cfg{
-            replication_synchro_quorum = 3,
+            replication_synchro_quorum = 2,
             replication_synchro_timeout = 0.001,
         }
         t.assert_error_msg_content_equals(
             'Quorum collection for a synchronous transaction is timed out',
             box.space.test.insert, box.space.test, {1})
         box.cfg{
-            replication_synchro_quorum = 2,
+            replication_synchro_quorum = 1,
             replication_synchro_timeout = 100000,
         }
         box.space.test:insert{2}
```